extendr-api/tests/serializer_tests.rs

```diff
@@ -0,0 +1,58 @@
+#[cfg(feature = "serde")]
+mod test {
+    use extendr_api::prelude::*;
+    use extendr_api::serializer::to_robj;
+    use serde::Serialize;
+
+    ////////////////////////////////////////////////////////////////////////////////
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_serialize_struct() {
+        test! {
+            #[derive(Serialize)]
+            struct Test {
+                int: i32,
+                seq: Vec<&'static str>,
+            }
+
+            let test = Test {
+                int: 1,
+                seq: vec!["a", "b"],
+            };
+
+            let expected = list!(int=1, seq=list!("a", "b"));
+            assert_eq!(to_robj(&test).unwrap(), Robj::from(expected));
+        }
+    }
```

Comment:
Perhaps I do not fully understand `serde` or our marshaling rules, but why `Vec<&'static str>` is serizlized as `list("a", "b")` and not `c("a", "b")`?

Reply:
Serde has its own marshalling rules and we don't get to see the type until the first array element arrives.

As a result, we collect everything as Robs and lists.

We could perform and extra step to convert lists of identical types to their vector types, but it is never
going to be as efficient as converting with a priori knowledge of the type.

We can look at this transformation as a later step.

Reply:
Then I suggest writing it down somewhere -- in the docs or comments, for people like myself -- that if on the Rust side we work with `Vec<T>`, on the R side we should expect `list()`.